doc/chapters/debugging/index.rst

```diff
@@ -0,0 +1,197 @@
+
+GDB
+---
+
+Attaching a `gdb` session can be done in two ways, one is to run the `gdb` session
+and attach inside the container, the other is to attach to the process from the
+outside of the container. Using both methods have their own set of problems you
+need to solve.
+
+Running GDB inside container
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you run `gdb` from inside the container using `lxc-attach`, you should not
+need to change any paths etc to find the correct libraries etc. You will need
+to have the `PID` from inside the container available to connect to however. this
+can be found out by running::
+
+    sudo lxc-attach --name SC-X ps aux
+
+Then running::
+
+    sudo lxc-attach --name SC-X -- gdb --pid YY
+
+Replace YY with the `PID` as seen inside the container. You can also run the
+application directly inside the container and start it with gdb directly::
+
+    sudo lxc-attach --name SC-X -- gdb --args /bin/ls
+
+TODO: Uncertain why, but this fails with unable to find executable.
+
+Running GDB outside container
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you run `gdb` from outside the container you will run into some problems with the
+paths as the application is executing inside a `chroot` and `gdb` is executing
+outside of it, this is fairly easily fixed however. First off you need to
+provide `gdb` with the path of the executable as the path provided by the system
+is inside the chroot::
+
+    # ps aux |grep main
+    # sudo gdb --pid XXX /tmp/container//SC-0/gateways/app/main
+
+Begin with finding the global namespace `PID` of the application you are
+debugging and then attach `gdb` to that `PID`, and the path to the application
+in the `un-chrooted` filesystem.
+
+The application should now be possible to debug as normal. Most of the standard
+libraries etc are mounted inside the `chroot` from the outside and should be
+available as normal without any changes. If you have dynamic libraries inside
+your container that the application are linking to, you may need to add the
+proper `solib-search-paths` to the libraries.
+
+Debugging using QtCreator inside a container
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Debugging applications running inside a container can be done, however, all
+processes are run as root so you need to start `QtCreator` as root to be
+able to connect to the process inside the container.
+
+1. Start a container
+2. Start `QtCreator` as root (sudo)
+3. Open a project in `QtCreator` 
+4. Build the code into a runnable binary
+5. Run the binary program inside the container
+6. If you are using `DBUS`, a `PID` is returned, otherwise find it in the outside
+   system.
+7. Open *Debug -> Start Debugging -> Attach to Running Application...*
+8. Find the right `PID` in the list, and attach.
+9. The process will be halted and you can debug it from `QtCreator`.
+
+
+Core dumps
+----------
+
+Nothing special should be needed to debug a core file using for example
+`QtCreator` or `GDB`. Load the file as normal.
```

Comment:
I did generate a core dump and debugged it... basically, it works as you would expect, nothing special needed :)